lib/alchemy/config_loader.rb

```diff
@@ -0,0 +1,69 @@
+module Alchemy
+  class ConfigLoader
+    attr_reader :which
+
+    def initialize(which)
+      @which = which.sub(/\.yml$/, '')
+    end
+
+    # Reads the config file named +<which>.yml+ from +config/alchemy/+ folder.
+    #
+    def load_all
+      if paths.empty?
+        raise LoadError, "Could not find #{which}.yml file! Please run `rails generate alchemy:scaffold`"
+      else
+        # OPTIMIZE: remove duplicates?
+        paths.map(&method(:load_file)).reduce(&method(:merge))
```

Comment:
> I let you (re)implement the map-reduce (this why the rename, see COMMITMSG) in a more understandable way, because I don't have any clue how to do that without sacrificing expressiveness IMO.

No, I don't reimplement anything you want to have in core. Maintaining OpenSource projects is already a huge burden. Telling me to maintain even more code, is just not nice. Please rethink your attitude!